<launch>
  <arg name="Trainin_Cycles"        default="400"                         />
  <arg name="SLAM_Name"             default="gmapping"                    />
  <arg name="Robots_Quantity"       default="3"                           />
  <arg name="Robots_Launch_Name"    default="multi_robot_in_world.launch" />
  <arg name="SLAM_Package_Name"     default="slam_auto_calibrator"        />
  <arg name="SLAM_Launch_Name"      default="slam_launcher.launch"        />
  <arg name="APE_Topic_Name"        default="APE"                         />
  <arg name="Robots_Namespace_Base" default="tb3_"                        />
  <arg name="Ground_Truth_Filename" default="Ground_Truth_Image.png"      />

  <!-- Paths  arguments -->
  <arg name="This_Node_src_Path"    default="/home/cerlabrob/catkin_ws/src/slam_auto_calibrator/src/"                              />
  <arg name="Maps_Path"             default="/home/cerlabrob/catkin_ws/src/slam_auto_calibrator/Maps/"                             />
  <arg name="Params_File_Path"      default="/home/cerlabrob/catkin_ws/src/turtlebot3/turtlebot3_slam/config/gmapping_params.yaml" />

  <!-- Calibrator parameters -->
  <param name="TrainingCycles"        type="int" value="$(arg Trainin_Cycles)"        />
  <param name="SLAMName"              type="str" value="$(arg SLAM_Name)"             />
  <param name="RobotsQty"             type="int" value="$(arg Robots_Quantity)"       />
  <param name="RobotsLauchName"       type="str" value="$(arg Robots_Launch_Name)"    />
  <param name="SLAMPackageName"       type="str" value="$(arg SLAM_Package_Name)"     />
  <param name="SLAMLaunchName"        type="str" value="$(arg SLAM_Launch_Name)"      />
  <param name="APE_Topic_Name"        type="str" value="$(arg APE_Topic_Name)"        />
  <param name="Robots_Namespace_Base" type="str" value="$(arg Robots_Namespace_Base)" />
  <param name="Ground_Truth_Filename" type="str" value="$(arg Ground_Truth_Filename)" />
  <param name="This_Node_src_Path"    type="str" value="$(arg This_Node_src_Path)"    />
  <param name="Maps_Path"             type="str" value="$(arg Maps_Path)"             />
  <param name="Params_File_Path"      type="str" value="$(arg Params_File_Path)"      />

  <!-- Launch SLAM auto-calibrator node -->
  <node pkg="slam_auto_calibrator" type="calibrator.py" name="slam_auto_calibrator"/>
</launch>
